### PR TITLE
8343001: Adjust XSLT and XPath Extension Function Property

### PR DESCRIPTION
**Pull request by JoeWang-Java**
Disables XSLT and XPath Extension Functions by default, setting jdk.xml.enableExtensionFunctions to false.

Adjusted tests accordingly.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change requires CSR request [JDK-8343002](https://bugs.openjdk.org/browse/JDK-8343002) to be approved
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8343001](https://bugs.openjdk.org/browse/JDK-8343001): Adjust XSLT and XPath Extension Function Property (**Enhancement** - P4)
 * [JDK-8343002](https://bugs.openjdk.org/browse/JDK-8343002): Adjust XSLT and XPath Extension Function Property (**CSR**)

### Reviewers
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**) Review applies to [6daac1ed](https://git.openjdk.org/jdk/pull/22504/files/6daac1ed0ca023f6407a0724d4c3d1d262ce8d1f)
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22504/head:pull/22504` \
`$ git checkout pull/22504`

Update a local copy of the PR: \
`$ git checkout pull/22504` \
`$ git pull https://git.openjdk.org/jdk.git pull/22504/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22504`

View PR using the GUI difftool: \
`$ git pr show -t 22504`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22504.diff">https://git.openjdk.org/jdk/pull/22504.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22504#issuecomment-2513321279)
</details>

### CODE REVIEW

**Review by LanceAndersen — APPROVED** [review]
Overall Looks good Joe

please see the comment below

File: src/java.xml/share/classes/com/sun/org/apache/xalan/internal/xsltc/runtime/ErrorMessages.java [hunk 1]
```diff
@@ -275,10 +275,16 @@ public Object[][] getContents()
...
         {BasisLibrary.UNALLOWED_EXTENSION_FUNCTION_ERR,
-        "Use of the extension function ''{0}'' is not allowed when the secure processing feature is set to true."},
+        "Use of the extension function ''{0}'' is not allowed when extension "
+              + "functions are disabled by the secure processing feature or "
+              + "the property ''jdk.xml.enableExtensionFunctions''. "
+              + "To enable extension functions, set ''jdk.xml.enableExtensionFunctions'' to ''true''."},
 
         {BasisLibrary.UNALLOWED_EXTENSION_ELEMENT_ERR,
-        "Use of the extension element ''{0}'' is not allowed when the secure processing feature is set to true."},
+        "Use of the extension function ''{0}'' is not allowed when extension "
```
Comment (RogerRiggs) [inline src/java.xml/share/classes/com/sun/org/apache/xalan/internal/xsltc/runtime/ErrorMessages.java:284]: Is it correct that "element" has been replaced by "function" here?

File: src/java.xml/share/classes/com/sun/org/apache/xalan/internal/xsltc/trax/TransformerFactoryImpl.java [hunk 2]
```diff
@@ -216,7 +215,7 @@ public PIParamWrapper(String media, String title, String charset) {
     /**
      * <p>State of secure processing feature.</p>
      */
-    private boolean _isNotSecureProcessing = true;
+    private boolean _isNotSecureProcessing = false;
```
Comment (RogerRiggs) [inline src/java.xml/share/classes/com/sun/org/apache/xalan/internal/xsltc/trax/TransformerFactoryImpl.java:218]: An unfortunate double-negative state for the feature.
(Makes it hard to understand and easy to get confused about).

File: test/jaxp/javax/xml/jaxp/libs/jaxp/library/JAXPTestUtilities.java [hunk 3]
```diff
@@ -60,6 +61,15 @@
...
  */
 public class JAXPTestUtilities {
+    public static String CLS_DIR = System.getProperty("test.classes");
+    public static String SRC_DIR = System.getProperty("test.src");
+    public static boolean isWindows = false;
+    static {
+        if (System.getProperty("os.name").contains("Windows")) {
+            isWindows = true;
+        }
+    };
```
Comment (RogerRiggs) [inline test/jaxp/javax/xml/jaxp/libs/jaxp/library/JAXPTestUtilities.java:71]: Could be a direct assignment and final:
```suggestion
    public static final boolean isWindows = System.getProperty("os.name").contains("Windows");
```

File: test/jaxp/javax/xml/jaxp/unittest/transform/XSLTFunctionsTest.java [hunk 4]
```diff
@@ -152,7 +159,7 @@ private String getDefaultOpposite() {
      * @param externalDoc Content of the external xml document
      * @param expectedResult Expected transformation result
      **/
-    @Test(dataProvider = "document")
+    //@Test(dataProvider = "document")
```
[the remark this answers is not in the document]
Comment (JoeWang-Java) [reply test/jaxp/javax/xml/jaxp/unittest/transform/XSLTFunctionsTest.java]: Good catch - it was commented out for debugging the other tests.

File: src/java.xml/share/classes/com/sun/org/apache/xalan/internal/xsltc/runtime/ErrorMessages.java (hunk 1 above)
Comment (JoeWang-Java) [reply src/java.xml/share/classes/com/sun/org/apache/xalan/internal/xsltc/runtime/ErrorMessages.java]: "element" it is. Fixed.

**Review by RogerRiggs — APPROVED** [review]
Thanks for the updates

**Pull request closed without merging**